### ISSUE TITLE
Wireframe printing (aka Low-Fi Printing)

### ISSUE THREAD

**Issue by epoxi** [alexrj/Slic3r#2274]
Well, maybe this it too much for asking, but the idea is brilliant!

![](http://www.3ders.org/images2014/WirePrint-3d-printed-wireframe-preview-8.jpg)

This video explains this better --> https://www.youtube.com/watch?v=Ea4V7kb2VsY

More info ---> http://stefaniemueller.org/WirePrint/

Cheers!
ePoxi

**Comment by nicksears** [comment]
We cant close both of these (https://github.com/alexrj/Slic3r/issues/3048) because they're duplicates of each other. The other was closed because it is a duplicate, but this would be a very good option/addition.

**Comment by ramkam** [comment]
Agreed. Though, the underlying topic could be to have an "easy" to use lib to implement homebrewed slicers. Kinda like opencv does for ... Cv. Just throwing ideas (yeah ok, i think it's worth it)

**Comment by lordofhyphens** [comment]
@Nicksears yeah I wouldn't close this for dupe, but I would close dupes of this :)

@Ramkam I would probably start by examining the perimeter code. It would seem like a natural place to start, tracing the edges of the triangles of the mesh.

**Comment by lordofhyphens** [comment]
We received a proposal from a GSoC student for this project.

**Comment by NateTG** [comment]
> usage of this for support or infill raises complicated issues about collision of extruder with non-wireframe printed parts...

It really doesn't seem like that should be particularly challenging.

**Comment by alexrj** [comment]
@cvrieos will work on this within GSoC (#3933).

**Comment by alexrj** [comment]
This was my initial attempt at it:
https://github.com/alexrj/Slic3r/blob/wireframe/utils/wireframe.pl

**Comment by lordofhyphens** [comment]
@curieos

**Comment by croisez** [comment]
is this wireframe feature abandoned?

**Comment by curieos** [comment]
I still have interest in working on the feature, it will probably get less of my attention when I do get back to work though.

**Comment by css911** [comment]
@alranel is this still open will like to work on it, for gsoc 2022

**Comment by supermerill** [comment]
If you want to work on it, to be able to be reused in the all-c++ codebase, I advise to work on the c++ side of the thing.
Imo, it's maybe possible to integrate an option next to the "spiral vase" (same kind of thing) that, as the spiral vase, force many settings to a certain value (no infill, etc). Then, in the slicing process, instead of doing the normal perimetergenerator, it switches to your new class & method for creating the 3D paths.

If you get the merill_merge branch, you have at your disposal 3DPath objects that you can add into the ExtrusionEntityCollection. If you add the collection to the object's 'perimeter' one, it will be printed by the gcode part of the slicer. Set "no_sort" in the collection if the order of the paths is important (that tag only pally for the current collection, not the one sorted inside, so tag each collection that shouln't be shuffled at gcode creation).

**Comment by css911** [comment]
Hi thanks for the reply, will checkout the merill_merge branch. one query I had apart from Freenode is there any other communication forum that our org utilizes?

**Comment by supermerill** [comment]
I have a discord, for the branch: 4TVUG4VRzP
